rust/osm2lanes/src/tag/osm.rs

```diff
@@ -0,0 +1,136 @@
+use crate::tag::{TagKey, Tags};
+
+pub const HIGHWAY: TagKey = TagKey::from("highway");
+
+pub enum Highway {
+    Road(Road),
+    Link(Road),
+    Bridleway,
+    BusGuideway,
+    Construction,
+    Corridor,
+    Cycleway,
+    Escape,
+    Footway,
+    LivingStreet,
+    Path,
+    Pedestrian,
+    Proposed,
+    Raceway,
+    Residential,
+    Service,
+    Steps,
+    Track,
+}
+
+pub enum Road {
+    Motorway,
+    Trunk,
+    Primary,
+    Secondary,
+    Tertiary,
+    Unclassified,
+    Unknown, // https://wiki.openstreetmap.org/wiki/Tag:highway%3Droad
+}
+
+impl std::fmt::Display for Road {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Road::Motorway => write!(f, "motorway"),
+            Road::Trunk => write!(f, "trunk"),
+            Road::Primary => write!(f, "primary"),
+            Road::Secondary => write!(f, "secondary"),
+            Road::Tertiary => write!(f, "tertiary"),
+            Road::Unclassified => write!(f, "unclassified"),
+            Road::Unknown => write!(f, "road"),
+        }
+    }
+}
+
+impl std::str::FromStr for Highway {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Ok(match s {
+            "motorway" => Self::Road(Road::Motorway),
+            "trunk" => Self::Road(Road::Trunk),
+            "primary" => Self::Road(Road::Primary),
+            "secondary" => Self::Road(Road::Secondary),
+            "tertiary" => Self::Road(Road::Tertiary),
+            "unclassified" => Self::Road(Road::Unclassified),
+            "road" => Self::Road(Road::Unknown),
+            "motorway_link" => Self::Link(Road::Motorway),
+            "trunk_link" => Self::Link(Road::Trunk),
+            "primary_link" => Self::Link(Road::Primary),
+            "secondary_link" => Self::Link(Road::Secondary),
+            "tertiary_link" => Self::Link(Road::Tertiary),
+            "bridleway" => Self::Bridleway,
+            "bus_guideway" => Self::BusGuideway,
+            "construction" => Self::Construction,
+            "corridor" => Self::Corridor,
+            "cycleway" => Self::Cycleway,
+            "escape" => Self::Escape,
+            "footway" => Self::Footway,
+            "living_street" => Self::LivingStreet,
+            "path" => Self::Path,
+            "pedestrian" => Self::Pedestrian,
+            "proposed" => Self::Proposed,
+            "raceway" => Self::Raceway,
+            "residential" => Self::Residential,
+            "service" => Self::Service,
+            "steps" => Self::Steps,
+            "track" => Self::Track,
+            _ => return Err(s.to_owned()),
+        })
+    }
+}
+
+impl std::fmt::Display for Highway {
```

Comment:
It would be great if we could specify these when we declare the enum and have a from/to-string automatically. I've used https://docs.rs/strum_macros/0.23.1/strum_macros/derive.EnumString.html before, but not entirely sure this lets you do both directions

Reply:
Fine by me, just a thought. Maybe one day I'll play with proc macros finally and make something like this, if strum indeed can't